### ISSUE TITLE
Custom endpoints and regions for all S3 compatible storage providers

### ISSUE THREAD

**Issue by life5ign** [jberends/duplicity_backup#4]
Great project!  It would be great to have the ability to use other S3 providers (wasabi, digitalocean, etc).  

Here is a guide on using boto3 to connect to wasabi, which could be generalized to other providers: https://wasabi-support.zendesk.com/hc/en-us/articles/115002579891-How-do-I-use-AWS-SDK-for-Python-boto3-with-Wasabi-

The important parameters when constructing the boto3 s3 client are the s3 endpoint, access token and key, and the region...but I don't see in their example, for the S3 (not IAM) examples, that they're entering the region when building the boto s3 client.  But in the boto3 docs, it looks like you can provide `region_name` as a parameter: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html

**Comment by jberends** [comment]
HI @life5ign and @denismatveev - I've just merged in the PR from @denismatveev to enable custom s3 buckets. I am working on fixing the tests and bringing it up to py 3.7 and higher and will release it thereafter as v1.2.0

**Comment by life5ign** [comment]
Excellent, this is great news @jberends, thank you for implementing this!

**Comment by jberends** [comment]
I have implemented it and testing it. I also have added additional features for better use including the option to pass `extra_args` to duplicity. See the (prelim) changelog on v1.2.0

I invite you to test it as well in your usecase and report if there are any problems you encounter. I did not test encryption to be honest ATM. Although there is a good guide on it here: https://www.digitalocean.com/community/tutorials/how-to-use-duplicity-with-gpg-to-back-up-data-to-digitalocean-spaces

You can locally install the this package as executable command with:

```bash
pip install git+https://github.com/jberends/duplicity_backup@main#egg=duplicity_backup_s3 --force --upgrade
```

## v1.2.0 (UNRELEASED)

This release is named "The Other S3 As Well"-Release. We now support other S3 storage providers, other than only the Amazon one. So now you can use digitalocean (tested) or even dropbox for that matter as a remote storage for your backups.

* :star: Added the option `extra_args` in the configuration yaml. These `extra_args` may be provided as a list in the yaml and they are passed down to duplicity. See the duplicity documentation for the exact use of them. (#8)
* :star: Added the option to provide the full remote uri in the yaml section `remote > uri`. This will circumvent the construction of the uri based on endpoint, bucket and path and ensures that when you know what you are doing a lot of different storage providers may be used. See the duplicity help documentation how to provide the remote uri correctly. (#8)
* :star: Added GPG encryption possibilities. Thanks to @denismatveev. (#5)
* :star: Added the ability to use custom S3 endpoints. Now it does not default to amazon S3 and you can use S3 compatible storage targets such as digitalocean (tested on Digital Ocean). Thanks to @denismatveev. (#5)
* :+1: Improved `init` command to add a flag `--quiet` for less chatty way of initialisation of the config file. (#8)
* :+1: Thorough spring cleanup. Dropped support for Python releases below 3.7. `Black`-ened, `pyupgrade`-ed, `isort`-ed the codebase. Fixed tests and added tests for newer python releases. Switched to dependabot for dependency management.  (#8)

**Comment by jberends** [comment]
released v1.2.0. to PyPI.
